browser/views.py

```diff
@@ -624,3 +624,71 @@ def attendee_create(request, course_id):
         'form': form,
     }
     return render(request, template, context)
+
+
+@staff_member_required
+def add_investigator_record(request, person_id):
+    person = get_object_or_404(Person, pk=person_id)
+    template = "browser/investigator_create.html"
+    form = InvestigationForm()
+
+    if request.method == 'POST':
+        if 'SaveButton_Clicked' in request.POST:
+            form = InvestigationForm(request.POST, instance=person)
+            if form.is_valid():
+                investigator = Investigator(subject=form.cleaned_data.get('subject'),
+                                            role=form.cleaned_data.get('role'),
+                                            person_id=person_id, year=form.cleaned_data.get('year'),
+                                            changed_by=request.user)
+                investigator.save()
+                return HttpResponseRedirect(reverse('person', args=(person.id,)))
+
+        if 'CancelButton_Clicked' in request.POST:
+            return HttpResponseRedirect(reverse('person', args=(person.id,)))
+
+    context = {
+        'form': form,
+        'person': person,
+    }
+
+    return render(request, template, context)
+
+
+@staff_member_required
+def edit_investigator_record(request,person_id,research_id):
+
+    person = get_object_or_404(Person, pk=person_id)
+    research = get_object_or_404(Investigator, pk=research_id)
+    template = "browser/investigator_edit.html"
+
+    # Retreive investigator data already stored in database and display to the user
+    form = InvestigationForm(initial={'subject': research.subject, 'role': research.role,
+                                             'year': research.year})
+
+    if request.method == 'POST':
+        if 'UpdateButton_Clicked' in request.POST:
+            form = InvestigationForm(request.POST, instance=person)
+            if form.is_valid():
+                research.subject = form.cleaned_data.get('subject')
+                research.role = form.cleaned_data.get('role')
+                research.year = form.cleaned_data.get('year')
+                research.save()
+                return HttpResponseRedirect(reverse('person', args=(person.id,)))
+
+        if 'CancelButton_Clicked' in request.POST:
+            return HttpResponseRedirect(reverse('person', args=(person.id,)))
+
+    context = {
+        'form': form,
+        'person': person,
+        'investigator_data': research,
+    }
+
+    return render(request, template, context)
```

Comment:
create and update can be one function

Reply:
Is it alright if I keep create and update as separate ? 
This is because, update uses an extra parameter research-id which is not used in the create logic. And a research id cannot be passed as a parameter while creating a investigator data because it has not yet been created.